editor/quick_open_dialog.cpp

```diff
@@ -0,0 +1,980 @@
+
+		Ref<Texture2D> thumbnail;
+		float score = 0;
+	};
+
+	Vector<StringName> base_types;
+	Vector<Candidate> candidates;
+
+	String query;
+	int selection_index = -1;
+	int num_visible_results = 0;
+	int max_total_results = 0;
+
+	ContentDisplayMode content_display_mode = ContentDisplayMode::LIST;
+	Vector<QuickOpenResultItem *> result_items;
+
+	ScrollContainer *scroll_container = nullptr;
+	VBoxContainer *list = nullptr;
+	HFlowContainer *grid = nullptr;
+
+	PanelContainer *panel_container;
+	CenterContainer *no_results_container;
+	Label *no_results_label;
+
+	Label *file_details_path;
+	Button *display_mode_toggle;
+	CheckButton *include_addons_toggle;
+
+	OAHashMap<StringName, Ref<Texture2D>> file_type_icons;
+
+	void _create_initial_results(bool p_include_addons);
+	void _find_candidates_in_folder(EditorFileSystemDirectory *p_directory, const bool p_include_addons);
+
+	int _sort_candidates(const String &p_query);
+	void _update_result_items(const int p_new_visible_results_count, int p_new_selection_index);
+
+	void _move_selection_index(const Key p_key);
+	void _select_item(int p_index);
+
+	void _item_input(const Ref<InputEvent> &p_ev, const int p_index);
+
+	void _set_display_mode(const ContentDisplayMode p_display_mode);
+	void _toggle_display_mode();
+	void _toggle_include_addons(bool p_pressed);
+
+	static void _bind_methods();
+};
+
+class QuickOpenResultGridItem : public VBoxContainer {
+	GDCLASS(QuickOpenResultGridItem, VBoxContainer)
+
+public:
+	QuickOpenResultGridItem();
+
+	void set_content(const Ref<Texture2D> &p_thumbnail, const String &p_file_name);
+	void reset();
+	void highlight_item(bool p_enabled);
+
+private:
+	TextureRect *thumbnail;
+	Label *name;
+};
+
+class QuickOpenResultListItem : public HBoxContainer {
+	GDCLASS(QuickOpenResultListItem, HBoxContainer)
+
+public:
+	QuickOpenResultListItem();
+
+	void set_content(const Ref<Texture2D> &p_thumbnail, const String &p_file_name, const String &p_file_directory);
+	void reset();
+	void highlight_item(bool p_enabled);
+
+protected:
+	void _notification(int p_what);
+
+private:
+	static const int CONTAINER_MARGIN = 8;
+
+	MarginContainer *image_container;
+	VBoxContainer *text_container;
+
+	TextureRect *thumbnail;
+	Label *name;
+	Label *path;
+};
+
+class QuickOpenResultItem : public HBoxContainer {
+	GDCLASS(QuickOpenResultItem, HBoxContainer)
+
+public:
+	QuickOpenResultItem();
+
+	void set_content(const Ref<Texture2D> &p_thumbnail, const String &p_file_name, const String &p_file_directory);
+	void set_display_mode(ContentDisplayMode display_mode);
+	void reset();
+
+	void highlight_item(bool p_enabled);
+
+protected:
+	void _notification(int p_what);
+
+private:
+	QuickOpenResultListItem *list_item;
+	QuickOpenResultGridItem *grid_item;
+
+	bool is_hovering = false;
+	bool is_selected = false;
+
+	void _set_enabled(bool p_enabled);
+};
+
+//------------------------- Dialog
+
+String get_dialog_title(const Vector<StringName> &p_base_types) {
```

Comment:
Avoid making global methods unless necessary.

Reply:
And make them `static` if you absolutely need to, to avoid namespace pollution